pkg/startup/startup_test.go

```diff
@@ -987,3 +989,23 @@ var _ = Describe("UT for GenerateIPv6ULAPrefix", func() {
 		Fail("random bits were all zeros")
 	})
 })
+
+var _ = DescribeTable("UT for extractKubeadmCIDRs",
+	func(cm *v1.ConfigMap, expectedIPv4 string, expectedIPv6 string, expectErr bool) {
+		v4, v6, err := extractKubeadmCIDRs(cm)
+		if expectErr {
+			Expect(err).To(HaveOccurred())
+		} else {
+			Expect(err).ToNot(HaveOccurred())
+		}
+		Expect(v4).To(Equal(expectedIPv4))
+		Expect(v6).To(Equal(expectedIPv6))
+	},
+	Entry("nil config map", nil, "", "", true),
+	Entry("empty config map", &v1.ConfigMap{}, "", "", false),
+	Entry("v4 only config map", &v1.ConfigMap{Data: map[string]string{"ClusterConfiguration": "podSubnet: 192.168.0.0/16"}}, "192.168.0.0/16", "", false),
+	Entry("dual v4 config map", &v1.ConfigMap{Data: map[string]string{"ClusterConfiguration": "podSubnet: 192.168.0.0/16,10.10.0.0/16"}}, "192.168.0.0/16", "", false),
+	Entry("v6 only config map", &v1.ConfigMap{Data: map[string]string{"ClusterConfiguration": "podSubnet: fdff:ffff:ffff:ffff:ffff::/80"}}, "", "fdff:ffff:ffff:ffff:ffff::/80", false),
+	Entry("dual v6 config map", &v1.ConfigMap{Data: map[string]string{"ClusterConfiguration": "podSubnet: fdff:ffff:ffff:ffff:ffff::/80,fdff:ffff:ffff:ffff:ffff::/80"}}, "", "fdff:ffff:ffff:ffff:ffff::/80", false),
```

Comment:
I think it's worth at least one test with the full configmap present and not just the podSubnet.

e.g.,

```
ClusterConfiguration": "apiServerCertSANs:\n- 35.223.159.180\n- 127.0.0.1\napiServerExtraArgs:\n  audit-log-path: /var/log/calico/audit/kube-audit.log\n  audit-policy-file: /etc/kubernetes/pki/audit-policy.yaml\n  authorization-mode: Node,RBAC\n  basic-auth-file: /etc/kubernetes/pki/basic_auth.csv\napiServerExtraVolumes:\n- hostPath: /var/log/calico/audit/\n  mountPath: /var/log/calico/audit/\n  name: calico-audit\n  pathType: DirectoryOrCreate\n  writable: true\napiVersion: kubeadm.k8s.io/v1alpha3\nauditPolicy:\n  logDir: /var/log/kubernetes/audit\n  logMaxAge: 2\n  path: \"\"\ncertificatesDir: /etc/kubernetes/pki\nclusterName: kubernetes\ncontrolPlaneEndpoint: \"\"\netcd:\n  local:\n    dataDir: /var/lib/etcd\n    image: \"\"\nimageRepository: k8s.gcr.io\nkind: ClusterConfiguration\nkubernetesVersion: v1.12.7\nnetworking:\n  dnsDomain: cluster.local\n  podSubnet: 192.168.0.0/16\n  serviceSubnet: 10.96.0.0/12\nunifiedControlPlaneImage: \"\"\n",
```
(This is what my cluster gives me for `kubectl get cm -n kube-system kubeadm-config`)